File: Project/dentist-microservice/DentistService/src/main/java/com/dentist/Service/service/DentistService.java
package com.dentist.Service.service;

import com.dentist.Service.model.Dentist;
import java.util.List;

public interface DentistService {

    Dentist registerDentist(Dentist dentist);
    Dentist updateDentist(Dentist dentist);
    List<Dentist> getAllDentist(Integer pageNumber, Integer pageSize);
    List<Dentist> getDentistList();
    List<Dentist> getAllDentistWithAppointment(Integer pageNumber, Integer pageSize);
    Dentist getDentistById(Long dentistId);
    boolean deleteDentist(Long dentistId);
}

File: Project/dentist-front/src/app/service/dentist.service.ts
import { Injectable } from '@angular/core';
import { userUrl, dentistUrl, patientUrl, addressUrl, appointmentUrl, billingUrl, managerUrl } from './helper';
import { HttpClient } from '@angular/common/http';
import { Dentist } from '../model/dentist';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class DentistService {

  private baseUrl = dentistUrl;
  constructor(private _httpClient:HttpClient) { }

  //load all dentist list
  public getAllDentist(page):Observable<Dentist[]>{
    return this._httpClient.get<Dentist[]>(`${this.baseUrl}/dentist/all?pageNumber=${page}`); 
  }

  //load all dentist list without pagination
  public getDentistListExcludePagination():Observable<Dentist[]>{
    return this._httpClient.get<Dentist[]>(`${this.baseUrl}/dentist/`); 
  }
  

  //get dentist by id
  getDentistById(dentistId):Observable<Dentist>{
    return this._httpClient.get<Dentist>(`${this.baseUrl}/dentist/${dentistId}`)
  }

  // save Dentist location
  saveDentist(inputData:Object){ 
    return this._httpClient.post(`${this.baseUrl}/dentist/`,inputData);
  }

  // update Dentist location
  updateDentist(inputData:Object){ 
    return this._httpClient.put(`${this.baseUrl}/dentist/`,inputData);
  }
    //delete the Dentist data
    deleteDentistById(dentistId:number):Observable<Boolean>{ 
      return this._httpClient.delete<Boolean>(`${this.baseUrl}/dentist/${dentistId}`);
    }


}
